packages/frontend-2/components/workspace/dashboard/Dashboard.vue

```diff
@@ -74,25 +67,29 @@ const showBillingAlert = computed(
       workspace.value?.plan?.status === WorkspacePlanStatuses.CancelationScheduled)
 )
 
-onResult((queryResult) => {
-  if (
-    queryResult.data?.workspaceBySlug.creationState?.completed === false &&
-    queryResult.data.workspaceBySlug.creationState.state &&
-    !hasFinalized.value &&
-    import.meta.client
-  ) {
-    hasFinalized.value = true
-    finalizeWizard(
-      queryResult.data.workspaceBySlug.creationState.state as WorkspaceWizardState,
-      queryResult.data.workspaceBySlug.id
-    )
-  }
+watch(
+  workspace,
+  (newWorkspace) => {
+    if (newWorkspace) {
+      if (
+        newWorkspace.creationState?.completed === false &&
+        newWorkspace.creationState.state &&
+        !hasFinalized.value &&
+        import.meta.client
+      ) {
+        hasFinalized.value = true
+        finalizeWizard(
+          newWorkspace.creationState.state as WorkspaceWizardState,
+          newWorkspace.id
+        )
+      }
 
-  if (queryResult.data?.workspaceBySlug) {
-    useHeadSafe({
-      title: queryResult.data.workspaceBySlug.name
-    })
-    validateCheckoutSession(queryResult.data.workspaceBySlug)
-  }
-})
+      useHeadSafe({
```

Comment:
this already existed before so u can leave it, but - https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?source=copy_link#2a7cf25d251e403fac1ab53946887102

the correct way would be to invoke this in root scope and point it to a ref, that u then change inside this watcher